src/Algebra/Construct/Pointwise.agda

```diff
@@ -0,0 +1,224 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- For each `IsX` algebraic structure `S`, lift the structure to the
+-- 'pointwise' function space `A → S`: categorically, this is the
+-- *power* object in the relevant category of `X` objects and morphisms
+--
+-- NB the module is parametrised only wrt `A`
+------------------------------------------------------------------------
+
+{-# OPTIONS --cubical-compatible --safe #-}
+
+module Algebra.Construct.Pointwise {a} (A : Set a) where
+
+open import Algebra.Bundles
+open import Algebra.Core using (Op₁; Op₂)
+open import Algebra.Structures
+open import Data.Product.Base using (_,_)
+open import Function.Base using (id; _∘′_; const)
+open import Level
+open import Relation.Binary.Core using (Rel)
+open import Relation.Binary.Bundles using (Setoid)
+open import Relation.Binary.Structures using (IsEquivalence)
+
+
+private
+
+  variable
+    c ℓ : Level
+    C : Set c
+    _≈_ : Rel C ℓ
+    ε 0# 1# : C
+    _⁻¹ -_ : Op₁ C
+    _∙_ _+_ _*_ : Op₂ C
+
+  lift₀ : C → A → C
+  lift₀ = const
+
+  lift₁ : Op₁ C → Op₁ (A → C)
+  lift₁ = _∘′_
+
+  lift₂ : Op₂ C → Op₂ (A → C)
+  lift₂ _∙_ g h x = (g x) ∙ (h x)
+
+  liftRel : Rel C ℓ → Rel (A → C) (a ⊔ ℓ)
+  liftRel _≈_ g h = ∀ x → (g x) ≈ (h x)
+
+
+------------------------------------------------------------------------
+-- Setoid structure: here rather than elsewhere? (could be imported?)
+
+module _ (isEquivalenceC : IsEquivalence _≈_) where
+
+  open IsEquivalence isEquivalenceC
+
+  isEquivalence : IsEquivalence (liftRel _≈_)
+  isEquivalence = record
+    { refl = λ {f} x → refl {f x}
+    ; sym = λ f≈g x → sym (f≈g x)
+    ; trans = λ f≈g g≈h x → trans (f≈g x) (g≈h x)
```

Comment:
Have to wonder if `lift₂ trans` would work here.

Reply:
Ah!?
Is this already defined under `Function.Equality`?

Reply:
I was more thinking along the lines of the funky combinators in [Function.Base](https://agda.github.io/agda-stdlib/master/Function.Base.html) though I notice that they are currently non-dependent.

Though, as you point out, this is defined in the library already - now in `Function.Relation.Binary.Setoid.Equality`.

Reply:
Re: `Function.Relation.Binary.Setoid.Equality`
Another case of implicit (there) vs. explicit (here) in the quantification over `x` in the definition of `_≈_` (there) and `liftRel _≈_` (here)... sigh. 
Plus, in order to use it I need to artificially impose the `Propositional` setoid structure on `A`... oh dear, and the definition of `isEquivalence` is in-lined there rather than lifted out... so perhaps some (downstream) refactoring might be in order there?
Not sure the additional complexity is worth it?

Reply:
Or should I make the quantification implicit here? Cf. #2382

Reply:
I certainly would experiment with implicit here.